### PR TITLE
security/acme-client: fix vault support, added vault token

### PR DESCRIPTION
**Pull request by themarek**
The vault token is needed to enable the storage of keys in vault. This was not considered so far.

https://github.com/opnsense/plugins/issues/3525 discusses the issue and the reporter suggested a solution in his [blog](https://blog.infoitech.co.uk/opnsense-bugfix-acme-plugin-upload-to-vault/#1-opnsense-acme-plugin-form). This is the second step of getting the solution implemented after acme.sh merged the first fix: https://github.com/acmesh-official/acme.sh/pull/5159 and opnsense merged the [release](https://github.com/acmesh-official/acme.sh/releases/tag/3.0.8) to make it available there.

With that pull request is now possible to use vault for storing certificates for opnsense.

### CODE REVIEW

File: security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml [hunk 1]
```diff
@@ -1592,6 +1592,11 @@
                     <mask>/^.{1,1024}$/u</mask>
                     <ValidationMessage>Should be a string between 1 and 1024 characters.</ValidationMessage>
                 </acme_vault_prefix>
+                <acme_vault_token type="TextField">
+                                    <Required>Y</Required>
+                                    <mask>/^.{1,1024}$/u</mask>
+                                    <ValidationMessage>Should be a string between 1 and 1024 characters.</ValidationMessage>
```
Comment (fraenki) [inline security/acme-client/src/opnsense/mvc/app/models/OPNsense/AcmeClient/AcmeClient.xml:1598]: The indention of these lines is off. You probably used tabs but should use spaces instead.

File: security/acme-client/src/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/AcmeVault.php [hunk 2]
```diff
@@ -45,6 +45,9 @@ public function prepare()
         if ((string)$this->config->acme_vault_kvv2 == 1) {
             $this->acme_env['VAULT_KV_V2'] = 1;
         }
+        if (!empty((string)$this->config->acme_vault_token)) {
+                    $this->acme_env['VAULT_TOKEN'] = (string)$this->config->acme_vault_token;
```
Comment (fraenki) [inline security/acme-client/src/opnsense/mvc/app/library/OPNsense/AcmeClient/LeAutomation/AcmeVault.php:49]: Same here, please fix the indention.

**Review by fraenki — CHANGES_REQUESTED** [review]
Looks good so far, only the indention needs to be fixed. :)

**Comment by themarek** [comment]
@fraenki thank you for catching that :-) That didn't look good. Fixed.

**Review by fraenki — APPROVED** [review]

**Comment by fraenki** [comment]
Merged, thanks!

**Pull request merged**